pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -459,6 +460,46 @@ func (cloud *CloudProvider) InsertDisk(ctx context.Context, project string, volK
 	}
 }
 
+func (cloud *CloudProvider) UpdateDisk(ctx context.Context, project string, volKey *meta.Key, existingDisk *CloudDisk, params common.ModifyVolumeParameters) error {
+	klog.V(5).Infof("Updating disk %v", volKey)
+	// hyperdisks are zonal disks
+	// pd-disks do not support modification of IOPS and Throughput
+	return cloud.updateZonalDisk(ctx, project, volKey, existingDisk, params)
+}
+
+func (cloud *CloudProvider) updateZonalDisk(ctx context.Context, project string, volKey *meta.Key, existingDisk *CloudDisk, params common.ModifyVolumeParameters) error {
+	specifiedIops := params.IOPS != nil && *params.IOPS != 0
+	specifiedThroughput := params.Throughput != nil && *params.Throughput != 0
+	if !specifiedIops && !specifiedThroughput {
+		return fmt.Errorf("no IOPS or Throughput specified for disk %v", existingDisk.GetSelfLink())
+	}
+	updatedDisk := &computev1.Disk{
+		Name: existingDisk.GetName(),
+	}
+	if params.IOPS != nil && *params.IOPS != 0 {
+		updatedDisk.ProvisionedIops = *params.IOPS
```

Comment:
Can you accumulate paths here as well (in an array), and just make one call to Paths()

Reply:
Done